src/test/java/com/heapwhisperer/metrics/jvm/NonAccumulatingGarbageCollectorMetricSetTest.java

```diff
@@ -105,6 +107,46 @@ public void testNameOfBackgroundUpdateThread() throws Exception {
         assertTrue(foundBackgroundUpdateThread());
     }
 
+    @Test
+    public void testUniqueMetricNames() throws Exception {
+        GarbageCollectorMetricSet garbageCollectorMetricSet = new GarbageCollectorMetricSet();
+        MetricRegistry metricRegistry = new MetricRegistry();
+        metricRegistry.registerAll(garbageCollectorMetricSet);
+        metricRegistry.registerAll(new NonAccumulatingGarbageCollectorMetricSet(garbageCollectorMetricSet, 5));
+    }
+
+    @Test
+    public void testMetricNamesSeconds() throws Exception {
+        GarbageCollectorMetricSet garbageCollectorMetricSet = new GarbageCollectorMetricSet();
+        MetricRegistry metricRegistry = new MetricRegistry();
+        metricRegistry.registerAll(new NonAccumulatingGarbageCollectorMetricSet(garbageCollectorMetricSet, 5000));
+        for (String name : metricRegistry.getNames()) {
+            if (!NonAccumulatingGarbageCollectorMetricSet.GC_THROUGHPUT_METRIC_NAME.equals(name)) {
+                Assert.assertTrue("Unexpected metric name: " + name, name.contains("last-5-seconds"));
```

Comment:
nit: Assuming the mangled metric names are part of a contract with the user, these assertions could be a little stronger - `contains` doesn't guarantee we didn't change how we generate metric names between releases.